client/src/pages/Admin.js

```diff
@@ -51,43 +60,94 @@ function AdminPage() {
 			}
 		}
 
+async function getAvailableModules(){
+  try {
+    const availableModule = await fetch("/api/listOfmodules");
+    const modulesData = await availableModule.json();
+    return modulesData;
+  } catch (error) {
+    console.error(error);
+    return "No Module is not avaialbe";
+  }
+}
+
+
+useEffect(() => {
+	getAvailableModules().then((modulesResult) => setModules(modulesResult));
+}, [selectedModuleType]);
```

Comment:
if we don't expect the list of module types to be changed very often, you can move this to be done only when we load the page (in the other useEffect)